src/main/java/com/minecolonies/coremod/tileentities/TileEntityWareHouse.java

```diff
@@ -370,16 +358,7 @@ private boolean isToolInHut(final String tool, @NotNull final AbstractBuilding r
                 @Nullable final TileEntity entity = world.getTileEntity(pos);
                 if(entity instanceof TileEntityChest)
                 {
-                    if(tool.equals(Utils.PICKAXE))
-                    {
-                        hasItem = InventoryUtils.isPickaxeInProvider(entity, requestingBuilding.getNeededPickaxeLevel(), requestingBuilding.getBuildingLevel());
-                    }
-                    else
-                    {
-                        hasItem = InventoryUtils.isToolInProvider(entity, tool, requestingBuilding.getBuildingLevel());
-                    }
-
-                    if(hasItem)
+                    if(InventoryUtils.isToolInProvider(entity, tool, requestingBuilding.getNeededPickaxeLevel(), requestingBuilding.getBuildingLevel()))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)